In [1]:
import sqlite3
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
%matplotlib inline

In [2]:
con = sqlite3.connect('take.db')

In [3]:
tiles = pd.read_sql("select * from cv_piece_raw", con)

In [4]:
tiles = tiles.set_index(["episode_id", "id"])

In [5]:
tiles.head()

r           g           b           h  \
episode_id id                                                        
r2.10      tile-13  143.137517  244.644860   11.174232   42.626836   
           tile-7    11.154438  241.869822  111.375740   72.049704   
           tile-5    11.483669  240.073662  110.711605   71.376650   
           tile-4    11.179123  143.223595  244.788758  102.109327   
           tile-9    11.769231  240.940760  111.261715   71.386384   

                             s           v  orientation        h_skew  \
episode_id id                                                           
r2.10      tile-13  241.875834  245.367156    32.541958  right-skewed   
           tile-7   240.817160  242.338462    22.245955     symmetric   
           tile-5   238.566366  240.884642     6.379068   left-skewed   
           tile-4   241.885732  245.515133    30.315793  right-skewed   
           tile-9   238.613616  242.626879    21.377890  right-skewed   

                           v_skew  num_edges       position       pos_x  \
episode_id id                                                             
r2.10      tile-13      symmetric          6    left center   59.333333   
           tile-7   bottom-skewed          8  center center  141.666667   
           tile-5   bottom-skewed          9    left center  101.666667   
           tile-4       symmetric          6   right center  534.333333   
           tile-9       symmetric          6    left center   99.000000   

                         pos_y  
episode_id id                   
r2.10      tile-13   73.777778  
           tile-7    72.444444  
           tile-5    80.444444  
           tile-4   124.888889  
           tile-9   178.666667

In [6]:
referent = pd.read_sql("select * from referent", con)

In [7]:
tiles_referent = pd.merge(referent, tiles, how="inner", left_on=["object", "episode_id"], right_on=["id", "episode_id"])

In [8]:
tiles_referent

episode_id   object           r           g           b           h  \
0        r6.163   tile-0  233.084091    0.143750  240.659659  147.222159   
1        r6.162   tile-1  195.261845   39.054239    0.000000    5.836658   
2        r6.161   tile-6    9.110448  240.550746    0.879104   56.974627   
3        r6.160   tile-1   94.209113   72.955248   72.955248    0.000000   
4        r6.165  tile-11  110.459885  204.237822   38.082378   46.014327   
...         ...      ...         ...         ...         ...         ...   
1044      r7.80   tile-3  181.124143   58.205075  243.850480  137.407407   
1045     r7.149  tile-11  213.425175  213.425175  213.425175    0.000000   
1046     r6.159  tile-11  106.577745  103.764186  101.917465   11.486367   
1047     r7.170  tile-11  135.053121   65.760956  239.049137  129.545817   
1048      r7.72   tile-3  235.990338  102.576950    0.000000   12.708075   

               s           v  orientation        h_skew         v_skew  \
0     252.088636  240.796591    -5.293050  right-skewed      symmetric   
1     247.846010  193.120948    -3.267609     symmetric      symmetric   
2     249.061194  240.691791    17.594086     symmetric  bottom-skewed   
3      56.328723   92.538649   -36.517355     symmetric  bottom-skewed   
4     202.752149  203.656877     8.334750   left-skewed  bottom-skewed   
...          ...         ...          ...           ...            ...   
1044  193.353909  244.899177    -1.268507     symmetric  bottom-skewed   
1045    0.000000  211.010490     3.359027     symmetric      symmetric   
1046   10.941046  106.179808    21.045604  right-skewed     top-skewed   
1047  181.557105  238.836653    37.670200     symmetric      symmetric   
1048  249.192547  234.034507   -30.920948   left-skewed      symmetric   

      num_edges       position       pos_x       pos_y  
0             8       left top  102.000000  428.000000  
1            12   right center  575.666667  125.777778  
2             8    left center   67.666667   75.555556  
3             8       left top  104.666667  431.111111  
4             8    left center  101.333333  130.666667  
...         ...            ...         ...         ...  
1044          6   right center  501.000000  124.444444  
1045         12      right top  538.333333  380.000000  
1046          6      right top  535.666667  436.000000  
1047          8  center center  140.000000  178.666667  
1048          9    left center  101.333333   77.777778  

[1049 rows x 15 columns]

In [9]:
words = pd.read_sql("select * from hand", con)

In [10]:
words.head()

episode_id  inc   word  start_time  end_time
0     r6.163    1   dann        0.07      0.68
1     r6.163    2  <sil>        0.68      1.68
2     r6.163    3  <sil>        1.68      1.78
3     r6.163    4    aus        1.78      2.06
4     r6.163    5    der        2.06      2.24

In [11]:
tiles_referent_words = pd.merge(tiles_referent, words, how="inner", on=["episode_id"])

In [12]:
rot = tiles_referent_words[tiles_referent_words.word == "rot"]
non_rot = tiles_referent_words[tiles_referent_words.word != "rot"]

In [13]:
np.average(rot["r"]), np.average(non_rot["r"])

(229.55487059802778, 133.91163363946285)

In [14]:
links = tiles_referent_words[tiles_referent_words.word == "links"]
non_links = tiles_referent_words[tiles_referent_words.word != "links"]

In [15]:
np.average(links["pos_x"]), np.average(non_links["pos_x"])

(102.9410480349345, 325.75011811513116)

In [16]:
len(set(words["word"]))
#set(words["word"])

383

In [17]:
blue_words = tiles_referent_words[tiles_referent_words.word == "blau"]
non_blue_words = tiles_referent_words[tiles_referent_words.word != "blau"]

np.average(blue_words["b"]), np.average(non_blue_words["b"]) 

(229.99495708553331, 104.8970970392987)

## Questions

-  For the episodes that have the word "rot" in them, what is the average r column value? How does it compare to all of the episodes that don't have the word "rot" in them?
        
              with "rot"         without "rot"
        (229.55487059802778, 133.91163363946285). 
        
        The r value averages are significantly different. A higher r value indicates a more red object and that is seen for episodes with the word "rot" in them.
        
- What is the average pos_x value for episodes that had the word "links" ? How does it compare to episodes that don't have the word "links" in them?

           with "links"     without "links"
       (102.9410480349345, 325.75011811513116). 
       
       The pos_x value averages are significantly different. A lower pos_x value indicates a more-left object and that is seen for episodes with the word "links" in them.
        
- What is the size of the vocabulary? (i.e., what is the set of words in the words dataframe?)

        383
        
- Choose another word in the vocabulary that is not a color. Which column do you think it will correspond to? Following what you did for "rot" and "links", can you see if your assumptions are correct?

        I chose the word "blau" which means blue in german. I would expect that the average value of the `b` column would be higher than all the other episodes that don't contain the word "blau". My assumption was correct. See the average b values for episodes with and without the word "blau".
        
             with "blau"       without "blau"
        (229.99495708553331, 104.8970970392987)